Src/Newtonsoft.Json.Tests/Issues/Issue1295.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Newtonsoft.Json.Tests.TestObjects;
+#if DNXCORE50
+using Xunit;
+using Test = Xunit.FactAttribute;
+using Assert = Newtonsoft.Json.Tests.XUnitAssert;
+#else
+using NUnit.Framework;
+#endif
+
+namespace Newtonsoft.Json.Tests.Issues
+{
+    [TestFixture]
+    public class Issue1295 : TestFixtureBase
+    {
+        [Test]
+        public void Test()
+        {
+            var fsharpStructRecord = new FSharpStructRecordWithDataMember("Hi there");
+            var asJson = JsonConvert.SerializeObject(fsharpStructRecord);
+            var deserialized = JsonConvert.DeserializeObject<FSharpStructRecordWithDataMember>(asJson);
+
+            Assert.AreEqual(fsharpStructRecord.Foo, deserialized.Foo);
+        }
+
+        [Test]
+        public void TestJsonSanity()
+        {
+            var fsharpStructRecord = new FSharpStructRecordWithDataMember("42");
+            var asJson = JsonConvert.SerializeObject(fsharpStructRecord);         
+
+            Assert.AreEqual(@"{""foo_field"":""42""}", asJson);
```

Comment:
This test fails https://github.com/JamesNK/Newtonsoft.Json/pull/2860/checks?check_run_id=13544125215

Reply:
Thanks, have to #ifdef the value for net2.0